### ISSUE TITLE
Unexpected errors validating styles in v25 that worked fine for a long time in maplibre gl js v5

### ISSUE THREAD

**Issue by dschep** [maplibre/maplibre-style-spec#1751]
**Describe the bug**
Some styles that have been in use for a long time are failing to validate with spec v25 with errors about mixing legacy filters and expressions. It's unclear if these expressions worked as intended in v5 or were silently misbehaving but weren't noticed by users.

**To Reproduce**
Validate a style containing a filter like this in v25 (this existed in a @stadiamaps style before they patched it):
```json
[
  "all",
  [
    "==",
    "$type",
    "LineString"
  ],
  [
    "all",
    [
      "==",
      "class",
      "rail"
    ],
    [
      "has",
      "service"
    ]
  ]
]
```

Concrete code example (for now, ohm might fix their style as stadia maps did)
```js
import { validateStyleMin, v8 } from "@maplibre/maplibre-gl-style-spec";

const resp = await fetch("https://www.openhistoricalmap.org/map-styles/main/main.json");
const style = await resp.json();

const errors = validateStyleMin(style, v8);
console.log(errors)
```

```
% node test.mjs
[
  ValidationError {
    message: 'layers[39].filter[2][1]: Mixing deprecated filter syntax with expression syntax is not supported. Replace ["in","name",""] with ["in",["get","name"],["literal",[""]]].'
  },
  ValidationError {
    message: 'layers[40].filter[2][1]: Mixing deprecated filter syntax with expression syntax is not supported. Replace ["in","surface","water"] with ["in",["get","surface"],["literal",["water"]]].'
  },
  ValidationError {
    message: 'layers[41].filter[2][1]: Mixing deprecated filter syntax with expression syntax is not supported. Replace ["in","surface","water"] with ["in",["get","surface"],["literal",["water"]]].'
  }
]
```

It can also easily be replicated using ultra, the dev version of which uses ml v6

* ml v6 - validation error - https://dev.overpass-ultra.us/#run&style=https://www.openhistoricalmap.org/map-styles/main/main.json
* ml v5 - no validation error - https://overpass-ultra.us/#run&style=https://www.openhistoricalmap.org/map-styles/main/main.json

**Expected behavior**
Styles that have worked for years shouldn't suddenly raise validation errors if they are correctly rendered.

**Additional context**

Here's the slack thread where I first reported concerns: https://osmus.slack.com/archives/C03TFH5NE83/p1781556507994529

**Comment by HarelM** [comment]
Thanks!!
Can you copy the relevant expression of OHM here too so that if they fix it it won't affect the ability to trace down the bug?

**Comment by 1ec5** [comment]
Based on the error message:

https://github.com/OpenHistoricalMap/map-styles/blob/21595c330860524c69fc0d67e9682ee1639b812f/historical/historical.json#L889-L893
https://github.com/OpenHistoricalMap/map-styles/blob/21595c330860524c69fc0d67e9682ee1639b812f/historical/historical.json#L918-L922
https://github.com/OpenHistoricalMap/map-styles/blob/21595c330860524c69fc0d67e9682ee1639b812f/historical/historical.json#L941-L945

**Comment by HarelM** [comment]
Looks like all 3 expressions are mixing legacy and non legacy expressions, I'm not sure how it affects rendering to be honest, but it is a potential issue I guess...
Cc: @eddy-geek

**Comment by HarelM** [comment]
@1ec5 as far as I understand, these expressions are not working as intended, so I think you should fix them regardless of the fix provided in the linked PR:
- #1752
